src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedSingleLineJavadocAndInvalidJavadocPosition.java

```diff
@@ -0,0 +1,222 @@
+
+  /**
+   * summary.
+   *
+   * @throws CheckstyleException if a problem occurs
+   */
+  void foo4() {}
+
+  /** An especially short bit of Javadoc. */
+  void foo5() {}
+
+  /** An especially short bit of Javadoc. */
+  void foo6() {}
+
+  // violation below 'Summary javadoc is missing.'
+  /**
+   * @inheritDoc
+   */
+  void foo7() {}
+
+  /** {@inheritDoc} */
+  void foo8() {}
+
+  // violation below 'Summary javadoc is missing.'
+  /**
+   * @customTag
+   */
+  void bar() {}
+
+  /**
+   * summary.
+   *
+   * <h1>Some header </h1>
+   *
+   * {@inheritDoc} {@code bar1} text
+   */
+  void bar2() {}
+
+  // violation below 'Summary javadoc is missing.'
+  /**
+   * @customTag <a> href="https://github.com/checkstyle/checkstyle/"</a>text
+   */
+  void bar3() {}
+
+  /** Single line Javadoc that references {@link String}. */
+  void bar4() {}
+
+  // violation below 'Summary javadoc is missing.'
+  /**
+   * @return in single line javadoc
+   */
+  int bar5() {
+    return 0;
+  }
+
+  /**
+   * summary.
+   *
+   * @return in multi line javadoc
+   */
+  int bar6() {
+    return 0;
+  }
+}
+
+// violation below 'Javadoc comment is placed in the wrong location.'
+/** invalid javadoc. */
+/** valid javadoc. */
+class ExtraInputInvalidJavadocPosition {
+  // violation above '.* ExtraInputInvalidJavadocPosition has to reside in its own source file.'
+  // violation below 'Javadoc comment is placed in the wrong location.'
+  /** invalid javadoc. */
+}
+
+/** valid javadoc. */
+/* ignore */
+class ExtraInputInvalidJavadocPosition2 {
+  // violation above '.* ExtraInputInvalidJavadocPosition2 has to reside in its own source file.'
+  // violation below 'Javadoc comment is placed in the wrong location.'
+  /** invalid javadoc. */
+  static {
+    /* ignore */
+  }
+
+  // violation below 'Javadoc comment is placed in the wrong location.'
+  /** invalid javadoc. */
+  /** valid javadoc. */
+  int field1;
+
+  /** valid javadoc. */
+  int[] field2;
+
+  /** valid javadoc. */
+  public int[] field3;
+
+  /** valid javadoc. */
+  @Deprecated int field4;
+
+  int
+      // violation below 'Javadoc comment is placed in the wrong location.'
+      /** invalid javadoc. */
+      field20;
+  int field21
+      // violation below 'Javadoc comment is placed in the wrong location.'
+      /** invalid javadoc. */
+      ;
+
+  @Deprecated
+  // violation below 'Javadoc comment is placed in the wrong location.'
+  /** invalid javadoc. */
+  JFrame frame = new JFrame();
+
+  void method1() {}
+
+  /** valid javadoc. */
+  void method2() {}
+
+  /** valid javadoc. */
+  <T> T method3() {
+    return null;
+  }
+
+  /** valid javadoc. */
+  String[] method4() {
+    return null;
+  }
+
+  void
+      // violation below 'Javadoc comment is placed in the wrong location.'
+      /** invalid javadoc. */
+      method20() {}
+
+  void method21
+      // violation below 'Javadoc comment is placed in the wrong location.'
+      /** invalid javadoc. */
+      () {}
```

Comment:
please move "// violation" comment out of target code

```
   // violation 2 line below 'Javadoc comment is placed in the wrong location.'
  void method21
      /** invalid javadoc. */
      () {}
```

do the same for:
![image](https://github.com/user-attachments/assets/abcc0c2a-884a-44f8-9b6a-4a0e9c702d7d)

Reply:
fixed, also changed violation msg location in some other places too, here's the new diff https://github.com/checkstyle/checkstyle/pull/15601#issuecomment-2327019594